x-pack/plugins/apm/typings/es_schemas/Span.ts

```diff
@@ -39,6 +39,15 @@ export interface Span extends APMDoc {
     subtype: string;
     sync: boolean;
     type: string;
+    http?: {
+      url: {
+        original: string;
+      };
+    };
+    db?: {
```

Comment:
Do you want to add the other values for `span.db` here?
https://github.com/elastic/apm-server/blob/master/_meta/ecs-migration.yml#L178-L196

Reply:
i usually add properties as needed, so there's less to manually maintain, but it might be worth it to mirror these types with the schema regardless of usage.

Reply:
it would be best if we could generate the types from the yaml spec

Reply:
Yeah, we _need_ some kind of definitive documentation for these schemas. I thought that was gonna be the types but one way or another.

Reply:
I think we should only add fields that we actually use - and fields from the interface when they are no longer used.

Previously I added fields that we didn't need. Since we don't yet have an integration test with APM Server some of these fields were changed (in APM Server) without us knowing (and we were not affected by it because we didn't use them). Result: we had stale fields.

There are also just a lot of fields we don't need, and it'll be a hassle to update them whenever they change, even if we don't use them or never intend to use them.

Lastly, sometimes APM Server folks ask, what whether it has consequences for the UI if they change a field. It is very easy to answer, if we can see that the fields is not in the interface, and we therefore know it is not used.

Reply:
That makes sense, but then we need a different way to document the schema.

Reply:
What kind of documentation do you have in mind? Isn't the interface self-documenting?